java/java-tests/testSrc/com/intellij/java/codeInspection/DataFlowInspection16WithMockTest.java

```diff
@@ -0,0 +1,24 @@
+// Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.intellij.java.codeInspection;
+
+import com.intellij.JavaTestUtil;
+import com.intellij.testFramework.LightProjectDescriptor;
+import org.jetbrains.annotations.NotNull;
+
+public class DataFlowInspection16WithMockTest extends DataFlowInspectionTestCase {
```

Comment:
I think it should be ok to add a new test to the existing `DataFlowInspection16Test` or `DataFlowInspection17Test` class. It's not so important that the particular test has library class mocks, as this doesn't affect other tests in the same TestCase class and doesn't require any specific setup steps. Also, I think that nothing bad will happen if mockJDK already has the Stream class and you redefine it inside the test. The references will be naturally resolved to the redefined one, as it's in the same file. In this case, you probably don't need to mock BaseStream at all.

Reply:
I moved test and created mock in this test-case.

Thanks again!